server/controllers/postController.ts

```diff
@@ -10,10 +10,14 @@ async function postPosts(req, res, next) {
 	const username = await userManager.findRefreshToken(refreshToken);
 	const { body: { title, body: postBody, parent, date } } = req;
 	if (date) {
-		scheduleJob({ date, title, body: postBody, author: username, parent }, function(postManager, argsObj: Partial<IPost>) {
-			postManager.addPost(argsObj.title, argsObj.body, argsObj.author, argsObj.parent);
+		const jobSuccess = await scheduleJob({ date, title, body: postBody, author: username, parent }, async function(postManager, argsObj: Partial<IPost>) {
+			await postManager.addPost(argsObj.title, argsObj.body, argsObj.author, argsObj.parent);
```

Comment:
You don't need to use async/await fo the callback. You use this structure not the first time. I would suggest to open some sandbox add try it for your self "what if" (what if add/remove/rewrite, what if error...). And perhaps try to rewrite it to simple promises (because async/await is just a syntax sugar). This will help you to understand it more. So this type of mistake won't affect this particular case, but it uncovers your knowledge about promises and async programming. So please take a deeper look into it.